examples/kitt/agent.py

```diff
@@ -0,0 +1,179 @@
+import asyncio
+from typing import List
+
+import dotenv
+from livekit import agents, rtc
+from livekit.agents import (
+    JobContext,
+    JobRequest,
+    WorkerOptions,
+    cli,
+)
+from livekit.agents.llm import ChatMessage, ChatRole, LLMStream
+from livekit.plugins.deepgram import STT
+from livekit.plugins.elevenlabs import TTS
+from livekit.plugins.openai import LLM
+
+dotenv.load_dotenv()
+
+
+async def entrypoint(job: JobContext):
+    # LiveKit Entities
+    chat_manager = rtc.ChatManager(job.room)
+    source = rtc.AudioSource(24000, 1)
+    track = rtc.LocalAudioTrack.create_audio_track("agent-mic", source)
+    options = rtc.TrackPublishOptions()
+    options.source = rtc.TrackSource.SOURCE_MICROPHONE
+
+    # Plugins
+    tts = TTS()
+    llm = LLM()
+    stt = STT()
+    stt_stream = stt.stream()
+
+    # State
+    committed_messages: List[agents.llm.ChatMessage] = [
+        ChatMessage(role=ChatRole.SYSTEM, text="You are a friendly assistant.")
+    ]
+    llm_stream_queue = asyncio.Queue[LLMStream]()
+    latest_llm_stream: LLMStream = None
+    tts_stream_queue = asyncio.Queue[agents.tts.SynthesizeStream]()
+    latest_tts_stream = tts.stream()
+    working_text = ""
+    agent_speaking = False
+
+    audio_stream_future = asyncio.Future[rtc.AudioStream]()
+
+    def on_track_subscribed(track: rtc.Track, *_):
+        if track.kind == rtc.TrackKind.KIND_AUDIO:
+            audio_stream_future.set_result(rtc.AudioStream(track))
+
+    def on_chat_message(message: rtc.ChatMessage):
+        print("Got chat message: ", message)
+
+    # IMPORTANT: Subscribe to audio tracks immediately. If you don't,
```

Comment:
are we using autosubscribe in this case? If so this comment wouldn't be relevant.

if we are not using autoSubscribe, hoping there are some mechanics built-in that can help us subscribe to tracks. feels like learning about this published thing is really confusing